dagger/config_finder/config_processor.py

```diff
@@ -40,20 +52,24 @@ def process_pipeline_configs(self):
             config_path = join(pipeline_config.directory, pipeline_config.config)
 
             _logger.info("Processing config: %s", config_path)
-            pipeline = Pipeline(pipeline_config.directory, self._load_yaml(config_path))
+            config_dict = self._load_yaml(config_path)
+            config_dict = self.overwrite_params(config_dict)
+            pipeline = Pipeline(pipeline_config.directory, config_dict)
 
             for task_config in pipeline_config.job_configs:
                 task_name = splitext(task_config.config)[0]
                 task_config_path = join(pipeline_config.directory, task_config.config)
 
                 _logger.info("Processing task config: %s", task_config_path)
-                task_config = self._load_yaml(task_config_path)
-                task_type = task_config["type"]
-                pipeline.add_task(
-                    self._task_factory.create_task(
-                        task_type, task_name, pipeline_name, pipeline, task_config
+                task_config = EnvYAML(task_config_path).export()
```

Comment:
What is the reason we are not using the EnvYAML for the pipeline config, only for the task config?